### PR TITLE
Use pathlib.Path in tests

### PR DESCRIPTION
**Pull request by eumiro**
Use `pathlib.Path` for directory/files manipulation in the tests. Remove the `relative_path_to` method.

No change outside of the `test` directory.

### CODE REVIEW

**Comment by eumiro** [comment]
Just discovered several unused imports. Fixed that, repushed.

File: test/barcodes/test_barcodes.py [hunk 1]
```diff
@@ -1,6 +1,10 @@
+from pathlib import Path
+
 from fpdf import FPDF
 from test.utilities import assert_pdf_equal
 
+HERE = Path(__file__).resolve().parent
```
Comment (Lucas-C) [inline test/barcodes/test_barcodes.py:6]: `PARENT_DIR` could be more explicit than just `HERE`.

This remarks applies to all other files

File: test/image/png_images/test_png_file.py [hunk 2]
```diff
@@ -1,39 +1,29 @@
...
-        images = [
-            relative_path_to(f)
-            for f in os.listdir(relative_path_to("."))
-            if f.endswith(".png") and os.path.basename(f) not in not_supported
-        ]
-        images.sort()
-
-        for image in images:
-            if os.path.basename(image) in not_supported:
-                with pytest.raises(FPDFException):
```
Comment (Lucas-C) [inline test/image/png_images/test_png_file.py:33]: This logic of testing that an exception was raised has been lost in your refactor

File: test/image/test_image_info.py [hunk 3]
```diff
@@ -1,61 +1,26 @@
...
+from pathlib import Path
 
 import fpdf
 
-from test.utilities import relative_path_to
+HERE = Path(__file__).resolve().parent
 
 
 class TestImageParsing:
-    def break_down_filename(self, image):
```
Comment (Lucas-C) [inline test/image/test_image_info.py:11]: I'd like to preserve this unused method as it documents the naming conventions of the PNG files in this directory

File: test/image/test_image_info.py [hunk 4]
```diff
@@ -1,61 +1,26 @@
...
 
-        short_keys = ["f", "h", "bpc", "w", "cs", "trns", "dp", "pal"]
-
-        with open(relative_path_to("image_info.json")) as f:
-            expected = json.load(f)
-
-        for info, image in zip(infos, images):
-            short_info = {k: v for k, v in info.items() if k in short_keys}
-            assert short_info == expected[image]
+        for path in sorted((HERE / "png_test_suite").glob("*.png")):
```
Comment (Lucas-C) [inline test/image/test_image_info.py:15]: 👍

**Comment by eumiro** [comment]
Sorry @Lucas-C I just saw you refactored the test_cells, so I rebased it and repushed. Hope your review won't break.

File: test/barcodes/test_barcodes.py (hunk 1 above)
Comment (eumiro) [reply test/barcodes/test_barcodes.py:6]: Yes, naming is difficult and I was quite happy with `HERE`. I understand `PARENT_DIR` as the directory above other directory, which is not what we need here. I'd leave it as `HERE`, but if you wish, feel free to add another commit with renaming.

File: test/image/png_images/test_png_file.py (hunk 2 above)
Comment (eumiro) [reply test/image/png_images/test_png_file.py:33]: Right, good point. Before we had:

```
images = [
    relative_path_to(f)
    for f in os.listdir(relative_path_to("."))
    if f.endswith(".png") and os.path.basename(f) not in not_supported
]
images.sort()

for image in images:
    if os.path.basename(image) in not_supported:
        with pytest.raises(FPDFException):
            pdf.image(x=0, y=0, w=0, h=0, link=None)
```

So there was `os.path.basename(f) not in not_supported` in the first list comprehension which eliminated all files that were in `not_supported`. So did that exception ever happen?

File: test/image/test_image_info.py (hunk 3 above)
Comment (eumiro) [reply test/image/test_image_info.py:11]: Okay, I changed it back.

File: test/image/png_images/test_png_file.py (hunk 2 above)
Comment (Lucas-C) [reply test/image/png_images/test_png_file.py:33]: Good point! Alright then

**Pull request merged**